Validation/MtdValidation/plugins/BtlDigiHitsValidation.cc

```diff
@@ -0,0 +1,279 @@
+  meHitEta_[0]    = ibook.book1D("BtlHitEtaL", "BTL DIGI hits #eta (L);#eta_{DIGI}", 100, -1.55, 1.55);
+  meHitEta_[1]    = ibook.book1D("BtlHitEtaR", "BTL DIGI hits #eta (R);#eta_{DIGI}", 100, -1.55, 1.55);
+
+
+  meHitTvsQ_[0]   = ibook.bookProfile("BtlHitTvsQL", "BTL DIGI ToA vs charge (L);Q_{DIGI} [ADC counts];ToA_{DIGI} [TDC counts]",
+				       50, 0., 1024., 0., 1024.);
+  meHitTvsQ_[1]   = ibook.bookProfile("BtlHitTvsQR", "BTL DIGI ToA vs charge (R);Q_{DIGI} [ADC counts];ToA_{DIGI} [TDC counts]",
+				       50, 0., 1024., 0., 1024.);
+  meHitQvsPhi_[0] = ibook.bookProfile("BtlHitQvsPhiL", "BTL DIGI charge vs #phi (L);#phi_{DIGI} [rad];Q_{DIGI} [ADC counts]",
+				      50, -3.15, 3.15, 0., 1024.);
+  meHitQvsPhi_[1] = ibook.bookProfile("BtlHitQvsPhiR", "BTL DIGI charge vs #phi (R);#phi_{DIGI} [rad];Q_{DIGI} [ADC counts]",
+				      50, -3.15, 3.15, 0., 1024.);
+  meHitQvsEta_[0] = ibook.bookProfile("BtlHitQvsEtaL","BTL DIGI charge vs #eta (L);#eta_{DIGI};Q_{DIGI} [ADC counts]",
+				      50, -1.55, 1.55, 0., 1024.);
+  meHitQvsEta_[1] = ibook.bookProfile("BtlHitQvsEtaR","BTL DIGI charge vs #eta (R);#eta_{DIGI};Q_{DIGI} [ADC counts]",
+				      50, -1.55, 1.55, 0., 1024.);
+  meHitQvsZ_[0]   = ibook.bookProfile("BtlHitQvsZL","BTL DIGI charge vs Z (L);Z_{DIGI} [cm];Q_{DIGI} [ADC counts]",
+				      50, -260., 260., 0., 1024.);
+  meHitQvsZ_[1]   = ibook.bookProfile("BtlHitQvsZR","BTL DIGI charge vs Z (R);Z_{DIGI} [cm];Q_{DIGI} [ADC counts]",
+				      50, -260., 260., 0., 1024.);
+  meHitTvsPhi_[0] = ibook.bookProfile("BtlHitTvsPhiL", "BTL DIGI ToA vs #phi (L);#phi_{DIGI} [rad];ToA_{DIGI} [TDC counts]",
+				      50, -3.15, 3.15, 0., 1024.);
+  meHitTvsPhi_[1] = ibook.bookProfile("BtlHitTvsPhiR", "BTL DIGI ToA vs #phi (R);#phi_{DIGI} [rad];ToA_{DIGI} [TDC counts]",
+				      50, -3.15, 3.15, 0., 1024.);
+  meHitTvsEta_[0] = ibook.bookProfile("BtlHitTvsEtaL","BTL DIGI ToA vs #eta (L);#eta_{DIGI};ToA_{DIGI} [TDC counts]",
+				      50, -1.55, 1.55, 0., 1024.);
+  meHitTvsEta_[1] = ibook.bookProfile("BtlHitTvsEtaR","BTL DIGI ToA vs #eta (R);#eta_{DIGI};ToA_{DIGI} [TDC counts]",
+				      50, -1.55, 1.55, 0., 1024.);
+  meHitTvsZ_[0]   = ibook.bookProfile("BtlHitTvsZL","BTL SIM ToA vs Z (L);Z_{DIGI} [cm];ToA_{DIGI} [TDC counts]",
+				      50, -260., 260., 0., 1024.);
+  meHitTvsZ_[1]   = ibook.bookProfile("BtlHitTvsZR","BTL SIM ToA vs Z (R);Z_{DIGI} [cm];ToA_{DIGI} [TDC counts]",
+				      50, -260., 260., 0., 1024.);
+
+}
+
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void BtlDigiHitsValidation::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+
+  edm::ParameterSetDescription desc;
+
+  desc.add<std::string>("folder", "MTD/BTL/DigiHits");
+  desc.add<std::string>("moduleLabel","mix");
+  desc.add<std::string>("btlDigiHitsCollection","FTLBarrel");
+
+  descriptions.add("btlDigiHits", desc);
```

Comment:
To anticipate premixing (#26332), please change this to
```suggestion
  descriptions.add("btlDigiHitsDefault", desc);
```
and create `Validation/MtdValidation/python/btlDigiHits_cfi.py` with
```python
import FWCore.ParameterSet.Config as cms
from Validation.MtdValidation.btlDigiHitsDefault_cfi import btlDigiHitsDefault as _btlDigiHitsDefault
btlDigiHits = _btlDigiHitsDefault.clone()

from Configuration.ProcessModifiers.premix_stage2_cff import premix_stage2
premix_stage2.toModify(btlDigiHits, moduleLabel = "mixData")
```